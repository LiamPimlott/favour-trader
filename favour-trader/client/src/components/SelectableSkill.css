.skillItem {
    padding: 2px;
    border-bottom-width: 1px;
    border-bottom-style: inset;
    border-bottom-left-radius: inherit;
}

.descriptionBox {
    width: 100%;
    border-radius: 2px;
    border: none;
}

.expansionTD {
    display: flex;
    text-align: center;
    position: fixed;
}

.checkboxTD {
    display: flex;
    text-align: center;
}

.checkbox {
    margin-top: 4px;
}