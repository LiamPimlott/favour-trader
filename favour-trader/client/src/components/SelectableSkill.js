import React, { Component } from 'react';
import { Icon } from 'antd';
import './SelectableSkill.css';

class SelectableSkill extends Component {
    constructor(props) {
        super(props);
        this.state = {
            selected: props.checked,
            expanded: true,
            description: '',
        };

        this.toggleExpansion = this.toggleExpansion.bind(this);
        this.toggleSelection = this.toggleSelection.bind(this);
    }

    toggleExpansion() {
        this.setState({
            expanded: !this.state.expanded,
        });
    }

    toggleSelection(e) {
        this.setState({
            selected: !this.state.selected,
            expanded: true,
        }, this.props.toggleSelection(e));
    }

    testSelect(e) {
        this.props.toggleSelection(e);
    }

    render() {
        const { skill, update, checked } = this.props;
        const { selected, expanded } = this.state;
        return (
            <tr className={'skillItem'}>
                <td className={'checkboxTD'}>
                    <input type={'checkbox'}
                        className={'checkbox'}
                        id={skill._id}
                        checked={checked}
                        onChange={this.toggleSelection}/>
                </td>
                <td>
                    {skill.skill}
                    {
                        (selected && expanded) ? (
                            <div>
                                {
                                    <textarea className={'descriptionBox'}
                                        id={skill._id}
                                        placeholder={'add a description (optional)'}
                                        onChange={update}/>
                                }
                            </div>
                        ) : (
                            ''
                        )
                    }
                </td>
                <td className={'expansionTD'}>
                    {
                    (selected) ? (
                        <Icon type={(expanded) ? ('up') : ('down')}
                            className='icon'
                            onClick={this.toggleExpansion}/>
                    ) : (
                        ''
                    )
                    }
                </td>
            </tr>
        );
    }
}

export default SelectableSkill;