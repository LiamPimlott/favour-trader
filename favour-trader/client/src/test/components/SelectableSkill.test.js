import React from 'react';
import renderer from 'react-test-renderer';
import SelectableSkill from '../../components/SelectableSkill.js';
import AuthServiceMock from './AuthServiceMock.js';

it('renders correctly', () => {
  var authServiceMock = new AuthServiceMock();
  const tree = renderer
    .create(<SelectableSkill key={123}
        skill={{
            _id: 123,
            category: {
              _id: 456,
              skill: 'plumber'

            }
        }}
        checked={true}
        update={() => {}}
        toggleSelection={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});