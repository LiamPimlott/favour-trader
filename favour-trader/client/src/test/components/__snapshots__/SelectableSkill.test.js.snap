// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  className="skillItem"
>
  <td
    className="checkboxTD"
  >
    <input
      checked={true}
      className="checkbox"
      id={456}
      onChange={[Function]}
      type="checkbox"
    />
  </td>
  <td>
    plumber
    <div>
      <textarea
        className="descriptionBox"
        id={456}
        onChange={[Function]}
        placeholder="add a description (optional)"
      />
    </div>
  </td>
  <td
    className="expansionTD"
  >
    <i
      className="anticon anticon-up icon"
      onClick={[Function]}
    />
  </td>
</tr>
`;
